catalog/action.go

```diff
@@ -0,0 +1,49 @@
+package catalog
+
+import (
+	"fmt"
+	"regexp"
+)
+
+type Action struct {
+	Name        string `yaml:"name"`
+	Description string `yaml:"description"`
+	On          struct {
+		PreMerge  *ActionOn `yaml:"pre-merge"`
+		PreCommit *ActionOn `yaml:"pre-commit"`
+	} `yaml:"on"`
+	Hooks []ActionHook `yaml:"hooks"`
+}
+
+type ActionOn struct {
+	Branches []string `yaml:"branches"`
+}
+
+type ActionHook struct {
+	ID          string            `yaml:"id"`
+	Type        string            `yaml:"type"`
+	Description string            `yaml:"description"`
+	Properties  map[string]string `yaml:"properties"`
+}
+
+var reHookID = regexp.MustCompile(`^[_a-zA-Z][_a-zA-Z0-9]{1,255}$`)
+
+func (a *Action) Validate() error {
+	if a.On.PreMerge == nil && a.On.PreCommit == nil {
+		return fmt.Errorf("%w 'on' is required", ErrInvalidAction)
+	}
+	ids := make(map[string]struct{})
```

Comment:
```suggestion
	ids := make(map[string]bool)
```
will give you a nicer usage:
```go
		if ids[hook.ID] {
			return fmt.Errorf("%w hook[%d] duplicate ID", ErrInvalidAction, i)
		}
```

Reply:
I prefer the zero size value